File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.py



class Solution:
    def nextGreatestLetter(self, stringArray, targetLetter):
        print("array: ", stringArray)
        
        for letter in stringArray:
            if (letter > targetLetter):
                return letter
        return stringArray[0]

instance=Solution()
print(instance.nextGreatestLetter(["c","f","j"], "c"))
print(instance.nextGreatestLetter(["x","x","y","y"],"z"))

File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.js


var nextGreatestLetter = (letterArray, targetLetter)=> {
let n=Math.min(...letterArray.map(letter => letter.charCodeAt()).filter(charCode=>charCode > targetLetter.charCodeAt()))

return n == Infinity? letterArray[0]: String.fromCharCode(n)

}

console.log(nextGreatestLetter(["c", "b"], "a"), "b", "the smallest letter thats greater than `a` in the array is `b`")
console.log(nextGreatestLetter(["m", "c"], "z"), "x", "return letterArray[0] since theres no letter greater than z")